Gems/PhysX/Code/Tests/EditorTestUtilities.cpp

```diff
@@ -323,4 +447,44 @@ namespace PhysXEditorTests
         // there should be no shapes on the rigid body because the cylinder radius and/or height is invalid
         EXPECT_EQ(pxRigidStatic->getNbShapes(), 0);
     }
+
+    void PhysXEditorFixture::ValidateInvalidEditorColliderComponentParams(const float radius, const float height)
+    {
+        // create an editor entity with a collider component
+        EntityPtr editorEntity = CreateInactiveEditorEntity("ColliderComponentEditorEntity");
+        auto* colliderComponent = editorEntity->CreateComponent<PhysX::EditorColliderComponent>();
+        editorEntity->CreateComponent<PhysX::EditorStaticRigidBodyComponent>();
+        editorEntity->Activate();
+
+        // Set collider to be a cylinder
+        AZ::EntityComponentIdPair idPair(editorEntity->GetId(), colliderComponent->GetId());
+        PhysX::EditorPrimitiveColliderComponentRequestBus::Event(
+            idPair, &PhysX::EditorPrimitiveColliderComponentRequests::SetShapeType, Physics::ShapeType::Cylinder);
+
+        {
+            UnitTest::ErrorHandler dimensionErrorHandler("SetCylinderRadius: radius must be greater than zero.");
+            PhysX::EditorPrimitiveColliderComponentRequestBus::Event(idPair, &PhysX::EditorPrimitiveColliderComponentRequests::SetCylinderRadius, radius);
+        
+            int expectedErrorCount = radius <= 0.f ? 1 : 0;
+            EXPECT_EQ(dimensionErrorHandler.GetExpectedErrorCount(), expectedErrorCount);
```

Comment:
comment: I'm not a huge fan of adding test expectations to member functions that get reused on other tests. It might be better if this is just tested in its own test, or these should be asserts. It's often much clearer for the expectation to be directly in the test so the failure is clearer from the test name and the context in which it failed

Reply:
That's a good comment, I agree. I followed the same pattern as in `ValidateInvalidEditorShapeColliderComponentParams` when created this function `ValidateInvalidEditorColliderComponentParams` beside it.

In the header it has a comment that this is to work around parameterized tests causing issues for this particular tests, so I thought to be safer to follow the same pattern:

````
        // workaround for parameterized tests causing issues with this (and any derived) fixture
        void ValidateInvalidEditorShapeColliderComponentParams(float radius, float height);
        void ValidateInvalidEditorColliderComponentParams(float radius, float height);
````

Maybe @SergeyAMZN or @greerdv can remember what was the issue with parameterized tests, and something that maybe can be improved in the future.

Reply:
I think it was @jonawals who added the cylinder tests.

Reply:
I think it's just to avoid copy-pasting the same code in 5 different tests. Could probably use parametrized test indeed.